bot/cogs/assignable_roles_cog.py

```diff
@@ -19,25 +21,139 @@ def __init__(self, bot) -> None:
+
+        # If RoleConverter failed
         except BadArgument:
-            await self.send_channel_list(ctx, f'@{input_role} not found')
+            # Casefold the roles
+            str_input_role = str(input_role).casefold()
+            role_list = ctx.guild.roles # Return list of roles
+            str_role_list = [str(i).casefold() for i in role_list] # Case-fold to do case insensitive matching
+
+            # Compare input_role to role_list entries for matches
+            role_count, matching_roles = 0, []
+
+            for j in range(len(str_role_list)):
+                if str_input_role == str_role_list[j]:
+                    #await self.set_role(ctx, role_list[j])
+                    matching_roles.append(role_list[j]) # matching_roles.append(j)
+                    role_count = role_count + 1
+            
+            if role_count == 0: # If no matches found, report findings
+                await self.send_role_list(ctx, f'@{input_role} not found')
+            elif role_count == 1: # If only one match was found, assign the role
+                await self.set_role(ctx, matching_roles[0]) # role_list[k[0]])
+            else: # If multiple matches found, query user via emojis to select correct role
+                await self.send_matching_roles_list(ctx, f'Multiple roles found for @{input_role}', \
+                    matching_roles, role_count)
+
+                #for m in range(len(k)):
+                #    await self.set_role(ctx, role_list[k[m]])
+
+    async def send_matching_roles_list(self, ctx, title: str, matching_roles, role_count):
+        #role_repo = RoleRepository()
+        #results = await role_repo.get_assignable_roles(ctx.guild.id)
+        names = ''
+        reactions = ['\u0031\ufe0f\u20e3','\u0032\ufe0f\u20e3','\u0033\ufe0f\u20e3','\u0034\ufe0f\u20e3', \
+                        '\u0035\ufe0f\u20e3','\u0036\ufe0f\u20e3','\u0037\ufe0f\u20e3','\u0038\ufe0f\u20e3', \
+                        '\u0039\ufe0f\u20e3','\U0001F51F']
+        """
+            USING EMOJIS WITH EMBEDDED TEXT
+                What I know works:
+                    1. Discord emoji name (e.g. ':pensive:')
+                    2. Unicode emoji name (e.g. '\u0031\ufe0f\u20e3' or '\U0001F3D3')
+        """
+        choose = '\n\n Please choose from one of the roles above.'
+
+        for k in range(len(matching_roles)):
+            if k < len(reactions):
+                emojis = reactions[k] + ' '
+            else:
+                emojis = ''
+            names = names + '\n' + emojis + str(matching_roles[k])
+        
+        embed = discord.Embed(title= title, color= Colors.ClemsonOrange)
+        embed.add_field(name= 'Matching Roles:', value= names + choose)
+
+        mes: Message = await ctx.send(embed= embed)
+        
+        # Remove unnecessary emojis
+        if role_count < len(reactions):
+            #print('\n' + str(len(reactions)) + '\n')
+            del reactions[role_count:len(reactions)]
```

Comment:
Yeppers.

Technically, I think that would need to be `reactions = reactions[:role_count-1]` because `role_count` is one-counting (length) while the `reactions` list is zero-counting